generator/src/main/java/com/scottlogic/deg/generator/restrictions/linear/LinearRestrictionsFactory.java

```diff
@@ -15,39 +16,39 @@
     }
 
     public static LinearRestrictions<OffsetDateTime> createDateTimeRestrictions(Limit<OffsetDateTime> min, Limit<OffsetDateTime> max, Timescale granularity) {
-        Limit<OffsetDateTime> cappedMin = capMin(min, ISO_MIN_DATE, ISO_MAX_DATE);
-        Limit<OffsetDateTime> cappedMax = capMax(max, ISO_MIN_DATE, ISO_MAX_DATE);
-        return new LinearRestrictions<>(cappedMin, cappedMax, granularity);
+        OffsetDateTime inclusiveMin = getInclusiveMin(min, granularity, ISO_MIN_DATE);
+        OffsetDateTime inclusiveMax = getInclusiveMax(max, granularity, ISO_MAX_DATE, b->b.minusNanos(100));
```

Comment:
Does this behave correctly for a field of granularity higher than the lowest (say, years) with a single `before` constraint? How about a `beforeOrAt` constraint?

Reply:
i should probably put a .previous() method on the granularity, and test that.